src/battle_ai_switch_items.c

```diff
@@ -28,7 +28,43 @@ static bool8 ShouldUseItem(void);
 static bool32 AiExpectsToFaintPlayer(void);
 static bool32 AI_ShouldHeal(u32 healAmount);
 static bool32 AI_OpponentCanFaintAiWithMod(u32 healAmount);
-static bool32 IsAiPartyMonOHKOBy(u32 battlerAtk, struct Pokemon *aiMon);
+EWRAM_DATA struct SwitchinCandidate switchinCandidate = {0};
+
+static void InitializeSwitchinCandidate(struct Pokemon *mon)
```

Comment:
This has a long answer, so in bullets:
- I needed to add a field to track the hypothetical status condition associated with checking TSpikes damage, and didn't want to mess with the BattlePokemon struct directly for such a niche application
- I thought making a new struct would make this code more easily extensible in future, again without needing to add to BattlePokemon all the time
- I'm only using about 2/3 of the fields in BattlePokemon, and assigning anything unnecessary quickly adds time to the function; my first draft of this I didn't realize I could just not assign the parameters I didn't care about
- I tried to have my struct have two elements, one being a pointer to a BattlePokemon struct and one being the hypothetical status, and while I could get it working in references in the switch AI functions, I just couldn't get it to work properly for GetBestMonAfterKOIntegrated. I'm not a C programmer and the rules / syntax involved in doing this correctly are just a bit beyond me, I spent another ~5ish hours today and had no better luck.

Certainly open to making the last bullet point change, it's just beyond my ability unfortunately :)

Reply:
In this case I would create a struct with fields like these:
```
struct AiSwitchMon 
{
    struct BattlePokemon mon;
    // all of the additional fields which the BattlePokemon struct does not have
}
```

So instead of passing a pointer you would just copy the whole struct at once which would be both more readable and faster. 
`dst->x = switchinCandidate->x` would simply become `dst = switchinCandidate.mon;`

Reply:
This worked great, didn't realize this was "allowed". Thanks a lot! I'll see about putting a commit through today with the remaining changes / clarifications :)